phosphor-auditlog/auditLogMgr.cpp

```diff
@@ -0,0 +1,110 @@
+#include "auditLogMgr.hpp"
+
+#include "auditLogParser.hpp"
+#include "auditLogUtils.hpp"
+
+#include <fcntl.h>
+#include <libaudit.h>
+
+#include <phosphor-logging/lg2.hpp>
+#include <sdeventplus/event.hpp>
+#include <sdeventplus/source/event.hpp>
+#include <xyz/openbmc_project/Common/File/error.hpp>
+
+#include <cstring>
+#include <string>
+
+namespace phosphor::auditlog
+{
+
+sdbusplus::message::unix_fd ALManager::getAuditLog()
+{
+#ifdef AUDITLOG_KEEP_JSONFILE
+    ALParseFile parsedFile("/tmp/auditLog.json");
+#else
+    ALParseFile parsedFile;
+#endif
+
+    // Initialize to parse full audit log
+    ALParseAll auditParser(parsedFile);
+
+    lg2::debug("Method GetAuditLog filePath={FILEPATH}", "FILEPATH",
+               parsedFile.getPath());
+
+    // Parse all the events
+    auditParser.doParse();
+
+    /* Get file descriptor to return.
+     * openParseFD() throws an error if it fails to open the file.
+     */
+    auto fd = openParseFD(parsedFile);
+
+    /* Schedule the fd to be closed by sdbusplus when it sends it back over
+     * D-Bus.
+     */
+    sdeventplus::Event event = sdeventplus::Event::get_default();
+    fdCloseEventSource = std::make_unique<sdeventplus::source::Defer>(
+        event, std::bind(&ALManager::closeFD, this, fd, std::placeholders::_1));
+
+    return fd;
+}
+
+sdbusplus::message::unix_fd ALManager::getLatestEntries(uint32_t maxEvents)
+{
+#ifdef AUDITLOG_KEEP_JSONFILE
+    ALParseFile parsedFile("/tmp/auditEntries.json");
+#else
+    ALParseFile parsedFile;
+#endif
+
+    lg2::debug("Method GetLatestEntries: {FILE} maxEvents: {MAXCOUNT}", "FILE",
+               parsedFile.getPath(), "MAXCOUNT", maxEvents);
+
+    ALParseLatest auditParser(maxEvents, parsedFile);
+
+    // Parse events up to maxEvents specified
+    auditParser.doParse();
+
+    /* Get file descriptor to return.
+     * openParseFD() throws an error if it fails to open the file.
+     */
+    auto fd = openParseFD(parsedFile);
+
+    /* Schedule the fd to be closed by sdbusplus when it sends it back over
+     * D-Bus.
+     */
+    sdeventplus::Event event = sdeventplus::Event::get_default();
+    fdCloseEventSource = std::make_unique<sdeventplus::source::Defer>(
+        event, std::bind(&ALManager::closeFD, this, fd, std::placeholders::_1));
+
+    return fd;
+}
+
+int ALManager::openParseFD(ALParseFile& parsedFile)
```

Comment:
Wouldn't this be 'const arg' better like?
```
int ALManager::openParseFD(const ALParseFile& parsedFile)
```

Reply:
Yes, I've made this change.